stdlib/asm/crypto/fri/frif2e2.masm

```diff
@@ -0,0 +1,288 @@
+    sub.2 dup                               #[add'-2,add'-2,t_d,e1,e0,p,d,poe,..] add' - 2 now points to the new C
+    movdn.7                                 #[add'-2,t_d,e1,e0,p,d,poe,add'-2,..]
+    push.0.0.0.0                            #[0,0,0,0,add'-2,t_d,e1,e0,p,d,poe,add'-2,..]
+    dup.4 add.1                             #[add'-1,0,0,0,0,add'-2,t_d,e1,e0,p,d,poe,add'-2,..]
+    mem_loadw                               #[0,0,a1,a0,add'-2,t_d,e1,e0,p,d,poe,add'-2,..]
+    push.0.0                                #[0,0,0,0,a1,a0,add'-2,t_d,e1,e0,p,d,poe,add'-2,..]
+    movup.6                                 #[add'-2,0,0,0,0,a1,a0,t_d,e1,e0,p,d,poe,add'-2,..]
+    mem_loadw                               #[C,a1,a0,t_d,e1,e0,p,d,poe,add'-2,..]
+    swapw.2                                 #[e0,p,d,poe,a1,a0,t_d,e1,C,add'-2,..]
+    swap.2                                  #[d,p,e0,poe,a1,a0,t_d,e1,C,add'-2,..]
+    movup.7                                 #[e1,d,p,e0,poe,a1,a0,t_d,C,add'-2,..]
+    movup.7                                 #[t_d,e1,d,p,e0,poe,a1,a0,C,add'-2,..]
+    movupw.2                                #[C,t_d,e1,d,p,e0,poe,a1,a0,add'-2,..]
+    movup.7                                 #[p,C,t_d,e1,d,e0,poe,a1,a0,add'-2,..]
+    movup.7                                 #[d,p,C,t_d,e1,e0,poe,a1,a0,add'-2,..]
+end
+
+#! The following procedure computes an iteration of FRI verification for a query.
+#! The stack starts in the following initial configuration [d,p,C,t_d,e1,e0,poe,a1,a0,..] where d is 
+#! the current domain size, p the current position, C is the commitment of current layer, t_d 
+#! its corresponding tree depth, (e1,e0) is the evaluation point, poe is the position in the exponent
+#! of the domain generator, (a1,a0) is the alpha to compute the next evaluation point.
+#! TODO: Check where/if some checks are needed at the beginning, like p < d (p&d are u32) and that 2^t_d == d 
+#! Input: #[d,p,C,t_d,e1,e0,poe,a1,a0,add',add_f..]
+#! Output: #[t_d-1,e1,e0,p%d/2,d/2,poe_sq,add',add_f..]
+export.verify_query_layer.1
+    u32unchecked_div.2          #[d/2,p,C,t_d,...]
+    dup                         #[d/2,d/2,p,C,t_d,...]
+    dup.2 swap                  #[d/2,p,d/2,p,C,t_d,...]
+    u32checked_divmod           #[p%d/2,?,d/2,p,C,t_d,...]
+    swapw                       #[C,p%d/2,?,d/2,p,t_d,...]
+    dup.4                       #[p%d/2,C,p%d/2,?,d/2,p,t_d,...]
+    dup.9                       #[t_d,p%d/2,C,p%d/2,?,d/2,p,t_d,...]
+    mtree_get  swapw            #[C,V',p%d/2,?,d/2,p,t_d,...]  V' are the digest of values V = (v3,v2,v1,v0)
+    dropw
+
+    # Unhashing the leaf values
+    push.4.0.0.0.0.0.0.0
+    adv_loadw
+    loc_storew.0
+    push.0.0.0.0
+    rpperm
+    swapw.3
+    movup.4
+    assert_eq
+    movup.3
+    assert_eq
+    movup.2
+    assert_eq
+    assert_eq
+    dropw
+    loc_loadw.0
```

Comment:
It would be great if for this (and other) sub-sections we'd include the following:

1. Initial and final state of the stack (doesn't need to be as detailed, as you have in other sections, just initial and final state is fine).
2. Number of cycles this section is supposed to take.

Also, one nit comment: it would be great to add spaces after commas (e.g., `[a, b, c, ...]` instead of `[a,bc,...]`) would improve readability (at least for me). Doesn't need to be done retroactively to the things already here, but would be great to do going forward.